core/trino-main/src/main/java/io/trino/sql/analyzer/OutputColumn.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.analyzer;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableSet;
+import io.trino.execution.Column;
+import io.trino.sql.analyzer.Analysis.SourceColumn;
+
+import javax.annotation.concurrent.Immutable;
+
+import java.util.Objects;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+@Immutable
```

Comment:
Does this annotation do anything or is it just information?

Reply:
It has some compile time checks I guess.

Reply:
I was under the impression it is rarely used but I was wrong. It is used throughout the project.